pkg_config.bzl

```diff
@@ -86,15 +88,28 @@ def _ignore_opts(opts, ignore_opts):
             remain += [opt]
     return remain
 
+def _symlink_no_duplicate(ctx, src, dest):
+    if not dest.exists:
+        ctx.symlink(src, dest)
+        return True
+    return False
+
 def _symlinks(ctx, basename, srcpaths):
     result = []
     root = ctx.path("")
     base = root.get_child(basename)
     rootlen = len(str(base)) - len(basename)
     for src in [ctx.path(p) for p in srcpaths]:
         dest = base.get_child(src.basename)
-        ctx.symlink(src, dest)
-        result += [str(dest)[rootlen:]]
+        # if the directory already exists then:
+        # 1) attempt to modify the name with the parent directory name
+        # 2) if that fails don't create the symlink at all
+        # we can sometimes skip symlink creation because of ocassional
+        # duplicates in pkg_config output
+        if dest.exists:
```

Comment:
Instead of just looking up 1 level in the directory tree, what do you think about generating unique identifiers from the complete srcpath? (For example: `s.replace("_", "__").replace("/", "_slash_")`?) This will be more robust to collisions in file paths.